services/provider/server/server.go

```diff
@@ -969,6 +986,223 @@ func (s *OCSProviderServer) GetStorageClaimConfig(ctx context.Context, req *pb.S
 
 }
 
+// SHIM for GetStorageClaimConfig to return response to 4.18 clients
+func (s *OCSProviderServer) proxyGetStorageClaimConfigBlock(
+	req *pb.StorageClaimConfigRequest,
+	storageCluster *ocsv1.StorageCluster,
+	consumerConfig util.StorageConsumerResources,
+	kubeResources []client.Object,
+) ([]*pb.ExternalResource, error) {
+
+	var extR []*pb.ExternalResource
+
+	for _, resource := range kubeResources {
+		gvk, err := apiutil.GVKForObject(resource, s.scheme)
+		if err != nil {
+			return nil, err
+		}
+
+		switch gvk {
+		case v1.SchemeGroupVersion.WithKind("Secret"):
+			secret := resource.(*corev1.Secret)
+			provSecName := consumerConfig.GetCsiRbdProvisionerSecretName()
+			nodeSecName := consumerConfig.GetCsiRbdNodeSecretName()
+			if secret.Name == provSecName || secret.Name == nodeSecName {
+				oldSecretFormat := map[string]string{}
+				for k, v := range secret.Data {
+					oldSecretFormat[k] = string(v)
+				}
```

Comment:
`secret.Data` is already a `map[string]string`. What is the reason to copy it into an new `map[string]string`?

Reply:
`secret.data` is a `map[string][]byte` `configmap.data` is `map[string]string`